library/rbtrees.pl

```diff
@@ -35,12 +35,12 @@
 :- module(rbtrees,
           [ rb_new/1,                   % -Tree
             rb_empty/1,                 % ?Tree
-            rb_lookup/3,                % +Key, -Value, +T
-            rb_update/4,                % +Tree, +Key, +NewVal, -NewTree
+            rb_lookup/3,                % +Key, ?Value, +Tree
```

Comment:
This should be -Value in SWI-Prolog's mode annotation.  Any - argument may be instantiated at call, resulting in the same behavior as `p(X), X = value`.   Things that should strictly be unbound at entry must be marked with `--Arg`

Reply:
Done.
I think there were some mistakes in the original code (before I modified it), so please review all the declarations, both in the `module` declaration and the `%!` comments (not just the ones that I've changed).

I'm leaving `+Key` as-is; I'd prefer `++Key` because variables in the keys can cause subtle bugs and I can't think of a use for them. (See my reply to your comment on line 154 for variables in `rb_lookup/3`.)